Makefile

```diff
@@ -35,6 +35,9 @@ test: clean fmt REBUILD
 test-long: clean fmt REBUILD
 	go test -v -race -tags=test -timeout=180s ./...
 
+test-tpool: clean fmt REBUILD
+	go test -v -race -tags=test -timeout=8s github.com/NebulousLabs/Sia/modules/transactionpool
+
```

Comment:
I didn't realize you could type `./transactionpool`. But I still think I'd rather have a make target for each of the modules (I added one for the wallet in the next PR). I also reorganized the makefile a bit.

It's not super clean but anyone who takes 10 or 20 minutes to figure out everything that's happening in the Makefile should be able to catch up. I find it a lot nicer to have `make test-tpool` than to need to remember what all the flags are, to set a timeout, to set the tags.

The timeout came in to play many times while I was fixing things up.